File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/api/command/atm/DepositMoneyCommand.java
package org.jmolecules.example.axonframework.domain.api.command.atm;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

public record DepositMoneyCommand(
  @TargetAggregateIdentifier
  String accountId,
  int amount
) {
  // empty
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/core/model/bankaccount/command/DepositMoneyCommand.kt
package org.jmolecules.example.axonframework.core.model.bankaccount.command

import org.jmolecules.architecture.cqrs.annotation.Command
import org.jmolecules.ddd.annotation.Association
import org.jmolecules.example.axonframework.core.model.bankaccount.AccountId
import org.jmolecules.example.axonframework.core.model.bankaccount.Amount

/**
 * Deposit money command.
 */
@Command(namespace = "axon.bank", name = "DepositMoneyCommand")
data class DepositMoneyCommand(
  @Association
  val accountId: AccountId,
  val amount: Amount
)
